### PR TITLE
[NCL-8851] Split Log Viewer toolbar and the log body into separate components

### PR DESCRIPTION
**Pull request by patrikk0123**

### CODE REVIEW

File: src/components/LogViewer/LogViewerBase.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  areLinesWrapped: boolean;
+  toolbar?: ReactNode;
+}
+
+interface IOnScrollProps {
+  scrollOffsetToBottom: number;
+  scrollUpdateWasRequested: boolean;
+}
+
+export const LogViewerBase = ({
```
Comment (DnsZhou) [inline src/components/LogViewer/LogViewerBase.tsx:25]: Maybe consider using `React.memo` for child component, but I will leave it to you to decide whether to use memo or not.

File: src/components/LogViewer/LogViewerToolbar.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  isStatic: boolean;
+  setIsPaused: (value: boolean) => void;
+  isFollowing: boolean;
+  setIsFollowing: (value: boolean) => void;
+  areLinesWrapped: boolean;
+  setAreLinesWrapped: (value: boolean) => void;
+  customActions?: ReactNode[];
+}
+
+export const LogViewerToolbar = ({
```
Comment (DnsZhou) [inline src/components/LogViewer/LogViewerToolbar.tsx:18]: Maybe consider using `React.memo` for child component, but I will leave it to you to decide whether to use memo or not.

**Review by DnsZhou — APPROVED** [review]
The split component looks good to me and verified from my local. I was just thinking maybe we can use memo for child components(see https://react.dev/reference/react/memo), but I will leave it to you to decide whether to use memo or not.

File: src/components/LogViewer/LogViewerBase.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (patrikk0123) [reply src/components/LogViewer/LogViewerBase.tsx:25]: i don't find that necessary, it's only internal component for the log viewer

File: src/components/LogViewer/LogViewerToolbar.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (patrikk0123) [reply src/components/LogViewer/LogViewerToolbar.tsx:18]: ditto

File: src/components/LogViewer/LogViewerBase.tsx (hunk 1 above)
Comment (patrikk0123) [reply src/components/LogViewer/LogViewerBase.tsx]: I think it is a good idea, I encapsulated it in the memo

**Pull request merged**